debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/transforms/partitions/ComputePartitionConf.java

```diff
@@ -0,0 +1,94 @@
+            .withDisplayName("List of the tables that activate the SMT")
+            .withType(ConfigDef.Type.LIST)
+            .withWidth(ConfigDef.Width.MEDIUM)
+            .withImportance(ConfigDef.Importance.MEDIUM)
+            .withDescription("Comma separated list of table names that will be considered for the transformation");
+
+    static final Field PARTITION_TABLE_FIELD_NAME_MAPPINGS_FIELD = Field.create(FIELD_TABLE_FIELD_NAME_MAPPINGS_CONF)
+            .withDisplayName("Table field mapping")
+            .withType(ConfigDef.Type.STRING)
+            .withValidation(ComputePartitionConf::isValidMapping)
+            .withWidth(ConfigDef.Width.MEDIUM)
+            .withImportance(ConfigDef.Importance.MEDIUM)
+            .withDescription("comma-separated list of colon-delimited table field pairs, e.g. products:name,orders:purchaser");
+
+    static final Field FIELD_TABLE_PARTITION_NUM_MAPPINGS_FIELD = Field.create(FIELD_TABLE_PARTITION_NUM_MAPPINGS_CONF)
+            .withDisplayName("Table number of partition mapping")
+            .withType(ConfigDef.Type.STRING)
+            .withValidation(ComputePartitionConf::isValidMapping)
+            .withWidth(ConfigDef.Width.MEDIUM)
+            .withImportance(ConfigDef.Importance.MEDIUM)
+            .withDescription("comma-separated list of colon-delimited table partition number pairs, e.g. products:2,orders:3");
+
+    static Map<String, String> parseMappings(List<String> mappings) {
+        final Map<String, String> m = new HashMap<>();
+        for (String mapping : mappings) {
+            final String[] parts = mapping.split(MAPPING_SEPARATOR);
+            if (parts.length != 2) {
+                throw new ComputePartitionException("Invalid mapping: " + mapping);
+            }
+            m.put(parts[0], parts[1]);
+        }
+        return m;
+    }
+
+    public static int isValidMapping(Configuration config, Field field, Field.ValidationOutput problems) {
+        List<String> values = config.getStrings(field, LIST_SEPARATOR);
+        try {
+            parseMappings(values);
+            return 0;
+        }
+        catch (Exception e) {
+        }
+        problems.accept(field, values, "list of colon-delimited pairs, e.g. <code>foo:bar,abc:xyz</code>");
+        return 1;
+    }
+
+    static Map<String, Integer> parseIntMappings(List<String> mappings) {
+        final Map<String, Integer> m = new HashMap<>();
+        for (String mapping : mappings) {
+            final String[] parts = mapping.split(":");
+            if (parts.length != 2) {
+                throw new ComputePartitionException("Invalid mapping: " + mapping);
+            }
+            try {
+                int value = Integer.parseInt(parts[1]);
+                m.put(parts[0], value);
+            }
+            catch (NumberFormatException e) {
+                throw new ComputePartitionException("Invalid mapping value: " + e.getMessage());
```

Comment:
Please instead of message pass the caught exception as the nested exception to keep the full stacktrace. Make sure it is so in all instances.